src/vs/platform/prompts/common/constants.ts

```diff
@@ -56,16 +72,13 @@ export const isPromptFile = (
 export const getCleanPromptName = (
 	fileUri: URI,
 ): string => {
-	assert(
-		isPromptFile(fileUri),
-		`Provided path '${fileUri.fsPath}' is not a prompt file.`,
-	);
-
-	// if a Copilot custom instructions file, remove `markdown` file extension
-	// otherwise, remove the `prompt` file extension
-	const fileExtension = (fileUri.path.endsWith(COPILOT_CUSTOM_INSTRUCTIONS_FILENAME))
-		? '.md'
-		: PROMPT_FILE_EXTENSION;
-
-	return basename(fileUri.path, fileExtension);
+	const filename = basename(fileUri);
+	if (filename.endsWith(PROMPT_FILE_EXTENSION)) {
+		return filename.slice(0, -PROMPT_FILE_EXTENSION.length);
+	} else if (filename.endsWith(INSTRUCTION_FILE_EXTENSION)) {
+		return filename.slice(0, -INSTRUCTION_FILE_EXTENSION.length);
+	} else if (filename === COPILOT_CUSTOM_INSTRUCTIONS_FILENAME) {
+		return filename.slice(0, -3);
+	}
```

Comment:
Same here, the `else` statements just add a bit of noise to the text:

```suggestion
	if (filename.endsWith(PROMPT_FILE_EXTENSION)) {
		return filename.slice(0, -PROMPT_FILE_EXTENSION.length);
	}
	
	if (filename.endsWith(INSTRUCTION_FILE_EXTENSION)) {
		return filename.slice(0, -INSTRUCTION_FILE_EXTENSION.length);
	}
	
	if (filename === COPILOT_CUSTOM_INSTRUCTIONS_FILENAME) {
		return filename.slice(0, -3);
	}
```